src/Functions/DateTimeTransforms.h

```diff
@@ -786,21 +786,21 @@ struct ToDayOfWeekImpl
 {
     static constexpr auto name = "toDayOfWeek";
 
-    static inline UInt8 execute(Int64 t, const DateLUTImpl & time_zone)
+    static inline UInt8 execute(Int64 t, UInt8 week_day_mode, const DateLUTImpl & time_zone)
```

Comment:
Just as a suggestion, what about replacing `UInt8 week_day_mode` by `WeekDayMode mode` and let callers pass an enum value? That way, we would not need function `check_week_day_mode`.

Reply:
UInt8 is added to keep the same style with other similar functions like toWeek.